src/Couchbase.Extensions.Caching/CouchbaseCache.cs

```diff
@@ -2,41 +2,49 @@
 using System.Threading.Tasks;
 using Couchbase.Core;
 using Microsoft.Extensions.Caching.Distributed;
-using Microsoft.Extensions.Internal;
 using Microsoft.Extensions.Options;
 
 namespace Couchbase.Extensions.Caching
 {
+    public interface ICouchbaseCache : IDistributedCache
+    {
+        IBucket Bucket { get; }
+
+        IOptions<CouchbaseCacheOptions> Options { get; }
+
+        /// <summary>
+        /// Gets the lifetime or expiration from the <see cref="DistributedCacheEntryOptions"/>. Only
+        /// sliding expiration is currently supported. If <see cref="DistributedCacheEntryOptions.SlidingExpiration"/>
+        /// if not set, then the <see cref="CouchbaseCacheOptions.LifeSpan"/> will be used. If it is empty then the
+        /// default lifespan of zero (0) will be used which is infinite expiration.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        TimeSpan GetLifetime(DistributedCacheEntryOptions options = null);
+    }
+
     /// <summary>
     /// A <see cref="IDistributedCache"/> implementation for Couchbase Server.
     /// </summary>
-    public class CouchbaseCache : IDistributedCache
+    public class CouchbaseCache : ICouchbaseCache
     {
         internal static readonly TimeSpan InfiniteLifetime = TimeSpan.Zero;
 
-        internal IBucket Bucket { get; }
-
-        internal IOptions<CouchbaseCacheOptions> Options { get; }
+        public IBucket Bucket { get; }
```

Comment:
Should probably put `///<inheritdoc />` on these to pull the docs from the interface.